### ISSUE TITLE
it's 3.6.1, but this issue is still exist

### ISSUE THREAD

**Issue by silravend** [tronprotocol/java-tron#2340]
Please answer these questions before submitting your issue. Thanks!

What did you do?
I freeze my TRX for another contract address, 4000 TRX for bandwidth, and 3000 TRX for engine, my address is TNi7QoSGheU2iKZR3VYTcJ1XwmNAZmw7G3, and the target contract address is TLvUe7hvZsvYyfmZ3NkTHrMBwtU4pXn83K.

What did you expect to see?
I wish I can unfreeze them.

What did you see instead?
I see "Unable to unfreeze TRX. This could be caused because the minimal freeze period hasn't been reached yet.", I don't know why, is that because the contract I executed selfdestruct.

Maybe it is because I call the 'killdestruct' function on the contract TLvUe7hvZsvYyfmZ3NkTHrMBwtU4pXn83K, so when I try to unfreeze the TRX, it returns:

await tronWeb.transactionBuilder.unfreezeBalance("ENERGY", "TNi7QoSGheU2iKZR3VYTcJ1XwmNAZmw7G3", "TLvUe7hvZsvYyfmZ3NkTHrMBwtU4pXn83K").catch(err => {
console.log(err);
})

class org.tron.core.exception.ContractValidateException : AcquiredDelegatedFrozenBalanceForEnergy[0] < delegatedEnergy[3000000000],this should never happen

**Comment by silravend** [comment]
someone said 

the fix will be released in next version, which is 3.6.0
--

**Comment by taihaofu** [comment]
Please wait the [proposal 18](https://tronscan.org/#/proposal/18) to be approved. It's in witness voting process now.

**Comment by jeancky** [comment]
@silravend  Hi, silravend. Since #18 proposal has been approved. Have you successfully unfrozen your trx?

**Comment by silravend** [comment]
I tried unfreeze from 'https://tronscan.org/#/account?from=tronlink&type=frozen', still the same message:
"Unable to unfreeze TRX. This could be caused because the minimal freeze period hasn't been reached yet."

and I tried write a JS file as follow:
```
await tronWeb.transactionBuilder.unfreezeBalance("ENERGY", "TNi7QoSGheU2iKZR3VYTcJ1XwmNAZmw7G3", "TLvUe7hvZsvYyfmZ3NkTHrMBwtU4pXn83K").catch(err => {
console.log(err);
})
```
return
`class org.tron.core.exception.ContractValidateException : AcquiredDelegatedFrozenBalanceForEnergy[0] < delegatedEnergy[3000000000]`

can you please give me some tips or instructions how to unfreeze it?

**Comment by renchenchang** [comment]
could you tell me the fullnode ip which your tronweb connected to?

**Comment by xiongzhend** [comment]
https://api.trongrid.io

**Comment by renchenchang** [comment]
@silravend 
Oh, the address TLvUe7hvZsvYyfmZ3NkTHrMBwtU4pXn83K was a contract address and it suicided and was deleted from db. Then  someone sent 10trx to TLvUe7hvZsvYyfmZ3NkTHrMBwtU4pXn83K and It became a normal account. But it did not save the energy your froze to it when it became a normal account. so you can’t unfreeze it. Will fix it.

**Comment by silravend** [comment]
thank you, and when it will be fixed?

**Comment by renchenchang** [comment]
Fix it in version 4.0

**Comment by shydesky** [comment]
> thank you, and when it will be fixed?

hi, the bug is fixed in 3.6.5 version. please check it when upgraded. it may be released on Oct, 12
